src/lightning/data/streaming/combined.py

```diff
@@ -17,6 +17,10 @@
 from torch.utils.data import IterableDataset
 
 from lightning.data.streaming.dataset import StreamingDataset
+from lightning.data.utilities.env import _WorkerEnv
+
+__NUM_SAMPLES_YIELDED__ = "__NUM_SAMPLES_YIELDED__"
+__SAMPLES__ = "__SAMPLES__"
```

Comment:
The variables themselves don't need to be private I think. You may want to be able to import them in other places in the code if needed and it's weird to import private variables 😄 

```suggestion
_NUM_SAMPLES_YIELDED_KEY = "__NUM_SAMPLES_YIELDED__"
_SAMPLES_KEY = "__SAMPLES__"
```